packages/toolpad-app/src/toolpad/AppEditor/QueryIcon.tsx

```diff
@@ -29,25 +41,16 @@ export default function QueryIcon({ id: iconId, mode, sx }: QueryIconProps) {
   const ModeIcon = modeIconMap.get(mode ?? '');
 
   return (
-    <div style={{ display: 'flex' }}>
+    <div style={{ display: 'flex', gap: DataSourceIcon ? 0.5 : 0 }}>
```

Comment:
```suggestion
    <div style={{ display: 'flex', gap: 0.5 }}>
```

The gap has no effect if there is only one child